### PR TITLE
Create handler example

### PR DESCRIPTION
**Pull request by kayhide**
I created an example to demonstrate how to implement and use a component which takes an event handler.

This example also shows how to make some props optional so that a user does not necessarily give a handler.

### CODE REVIEW

**Comment by spicydonuts** [comment]
It might be better to split this into two examples, one which focuses on events and one which demonstrates optional props. Combining makes both a little harder to parse imo. I think there might already be event handler examples, but the optional props trick is neat.

**Comment by kayhide** [comment]
Okay, I will split it into two examples of:
- event handler
- optional props

Yeah, actually I thought the same, an event handler example should be existing, because it is a very basic trick when using reactjs.
But I couldn't find any.
I hope, this example(s) would help people ;)

**Comment by kayhide** [comment]
@spicydonuts I updated and splitted it into two. Please take a look.

File: examples/handler/Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+all:
+	nix-shell --pure --run "yarn build"
```
Comment (spicydonuts) [inline examples/handler/Makefile:2]: I don't think we want to add nix as a dependency for this example

**Comment by spicydonuts** [comment]
The examples themselves look good, btw, thanks!

**Comment by kayhide** [comment]
@spicydonuts Hi, thanks for your review.
Okay, I will have a look.

Or if you want, you can take over this branch and wrap up things so that the examples follow the format.

To be honest, I have not successfully built the examples yet without using spago or yarn.

**Comment by spicydonuts** [comment]
Yeah, I can look at it when I get a chance

**Pull request closed without merging**